java/vector/src/test/java/org/apache/arrow/vector/TestListVector.java

```diff
@@ -1115,6 +1115,14 @@ public void testIsEmpty() {
     }
   }
 
+  @Test
+  public void testTotalCapacity() {
+    try (final ListVector vector = ListVector.empty("list", allocator)) {
+      vector.setInitialTotalCapacity(10, 100);
+      assertEquals(100, vector.getValueCapacity());
```

Comment:
```suggestion
      assertEquals(10, vector.getValueCapacity());
      assertEquals(10, vector.getDataVector().getValueCapacity());
```

Reply:
I revised the test as suggested in the comment that demonstrates use of allocateNewSafe().